pkg/deployment/deployment.go

```diff
@@ -163,73 +174,55 @@ func (p *PartRole) UnmarshalJSON(data []byte) (err error) {
 }
 
 type RWVolume struct {
-	Path          string   `json:"path"`
-	Snapshotted   bool     `json:"snapshotted,omitempty"`
-	NoCopyOnWrite bool     `json:"noCopyOnWrite,omitempty"`
-	MountOpts     []string `json:"mountOpts,omitempty"`
+	Path          string   `yaml:"path"`
+	Snapshotted   bool     `yaml:"snapshotted,omitempty"`
+	NoCopyOnWrite bool     `yaml:"noCopyOnWrite,omitempty"`
+	MountOpts     []string `yaml:"mountOpts,omitempty"`
 }
 
 type RWVolumes []RWVolume
 
 type Partition struct {
-	Label       string     `json:"label,omitempty"`
-	FileSystem  FileSystem `json:"fileSystem,omitempty"`
-	Size        MiB        `json:"size,omitempty"`
-	Role        PartRole   `json:"role"`
-	StartSector uint       `json:"startSector,omitempty"`
-	MountPoint  string     `json:"mountPoint,omitempty"`
-	MountOpts   []string   `json:"mountOpts,omitempty"`
-	RWVolumes   RWVolumes  `json:"rwVolumes,omitempty"`
-	UUID        string     `json:"uuid,omitempty"`
+	Label       string     `yaml:"label,omitempty"`
+	FileSystem  FileSystem `yaml:"fileSystem,omitempty"`
+	Size        MiB        `yaml:"size,omitempty"`
+	Role        PartRole   `yaml:"role"`
+	StartSector uint       `yaml:"startSector,omitempty"`
+	MountPoint  string     `yaml:"mountPoint,omitempty"`
+	MountOpts   []string   `yaml:"mountOpts,omitempty"`
+	RWVolumes   RWVolumes  `yaml:"rwVolumes,omitempty"`
+	UUID        string     `yaml:"uuid,omitempty"`
 }
 
 type Partitions []*Partition
 
 type Disk struct {
-	Device      string     `json:"device,omitempty"`
-	Partitions  Partitions `json:"partitions"`
-	StartSector uint       `json:"startSector,omitempty"`
+	// omit the device name as this is a runtime information which might
+	// not be consistent across reboots, there is no need to store it.
+	Device      string     `yaml:"-"`
```

Comment:
This forces setting the target device form the command line and not form config yaml.